File: DataStructures/LinkedList/RecursiveLength.cs


public class Node 
{
    public int data;
    public Node next
    public Node(int d)
    {
        data = d; 
        next = null;
    }


    public class LinkedList
    {
        Node head;
        public void push(int new_data)
        {
            Node new_node = new Node(new_data);
            new_node.next = head;
            head = new_node;
        }


        public int getCountRec(Node node)
        {
            if(node == null)
            {
                return 0;
            }
            else{
                return 1 + getCountRec(node.next);
            }
        }

        public int getCount(){
            return getCountRec(head);
        }

        public static void Main(String[] args)
        {
            LinkedList llist = new LinkedList();
            llist.push(1)
            llist.push(5)
            llist.push(4)
            llist.push(3)
            llist.push(2)

            Console.Write("Count of nodes is " + llist.getCount());
        }
    }
}

File: DataStructures/LinkedList/RecursiveLength.py
class Node:
    def __init__(self,data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None

    def push(self, new_data):
        new_node = Node(new_data)
        new_node.next = self.head
        self.head = new_node

    def getCountRec(self, node):
        if node is None:
            return 0
        else:
            return 1 + self.getCountRec(node.next)

    def getCount(self):
        return self.getCountRec(self.head)

    

#call functions with self in classes

if __name__=='__main__':
    llist = LinkedList()
    llist.push(1)
    llist.push(4)
    llist.push(3)
    llist.push(2)

    print('Count of node is: ', llist.getCount())